_transcripts/005-rust-1.38.0.md

```diff
@@ -0,0 +1,686 @@
+just like, you know, it looks like a normal, like, you know, syntax macro.
+
+__Jon__: Yeah. Exactly.
+
+__Ben__: The procedural, full blown shebang.
+
+__Jon__: Yeah, and all of these function-like macros are already generated by
+`rustdoc`, and they're listed normally in the documentation. But things like
+derived macros, and other kinds of attribute macros, such as— or attributes at
+all. Such as `deprecated` here, those, even though I don't know whether that's
+implemented as a macro internally in the compiler, but it is an attribute that
+you can use, and those I don't think are generated in the docs at all. And
+that's also why they don't show up in the documentation for the standard
+library.
+
+__Ben__: So that would be very nice to have, just for user convenience. Did I
+cut you off before? Did you want to say—
+
+__Jon__: No. I just wanted to say that this feels a bit to me like, similar
+change to being able to name macros in use statements, of just making macros
+feel more like first class citizens, alongside other things like types and
+traits and functions.
+
+The other thing I wanted to mention briefly, when we talk about deprecation, is
+that I've seen a number of crates use `#![deny(warnings)]` to make sure that,
+well, to essentially turn every warning into an error. This is a common pattern
+in other languages as well, where you use `-Wall -Werror` in C, for example. And
+in Rust, you have to be a little bit careful about doing this, because it means
+that even warnings about using deprecated methods also turn into hard errors.
+And this means that a semantic versioning compatible bump of a dependency might
+cause your crate to start issue warnings, which then become hard errors.
+
+__Ben__: If you have this deny—
+
+__Jon__: `#![deny(warnings)]`.
+
+__Ben__: You have all warnings denied.
+
+__Jon__: Yeah.
+
+__Ben__: I think even the Rust team themselves, like for the purpose of like,
+Crater runs— and so Crater is a tool that just checks new versions of the
+compiler against all the crates on crates.io to see if there are any, like,
+problems, and it actually ignores the deny warnings lint because of this, and so
+it's kind of like, you could just, if you want, enable this. It's not a big
+deal, like, deprecation is just a thing.
+
+__Jon__: And it's because it means that semantically, semver-compatible updates
+of dependencies might now break your build, even though the update was, in fact,
+completely semver-compatible.
+
+There's another thing that landed in 1.38 which is this `std::any::type_name`.
+So this is primarily used in debug context. So the idea here is that it's
+generic over some type `T` just like the `size_of` function, for example, and it
+just gives you a string that describes the type you're talking about. If you run
+`type_name`, generic over `bool`, then you'll get back a string that just says
+`"bool"`. And the idea is, you can use this to print out your types, if you
```

Comment:
Perhaps this should be formatted as `type_name::<bool>` ?

Reply:
No, I think sticking to how it was phrased is fine here.